src/harness/sas_objects.py

```diff
@@ -0,0 +1,353 @@
+#    Copyright 2016 SAS Project Authors. All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+import json
+import os
+import sas
+import urllib
+
+
+class Grant:
+
+    """
+    A Grant object has the following methods
+     __init__
+          Parameters: grant ID, grant request
+    Retrieve  grant ID,
+    Retrieve  grant request
+    Retrieve operation params
+    Construct heartbeat request
+    Update was authorized in last heartbeat? state
+    Retrieve was authorized in last heartbeat
+    """
+
+    def __init__(self, g_id, g_request):
```

Comment:
Please do not use abbreviations in the input variables -- there's no benefit to saving space and it's less readable this way.

Reply:
Abbreviations   are changed to meaningful names